Sources/Core/Other/PasteboardMonitor.swift

```diff
@@ -0,0 +1,76 @@
+//
+//  PasteboardMonitor.swift
+//  XestiMonitors-iOS
+//
+//  Created by Paul nyondo on 15/03/2018.
+//  Copyright © 2018 Xesticode. All rights reserved.
+//
+
+#if os(iOS)
+    import  UIKit
+    ///
+    /// A `PasteboardMonitor` instance monitors a pasteboard for changes to
+    /// its state.
+    ///
+    public class PasteboardMonitor: BaseNotificationMonitor {
+        
+        ///
+        /// Encapsulates changes to the contents of the pasteboard.
+        ///
+        
+        public enum Event {
+            ///
+            ///  Contents of the pasteboard object have changed.
+            ///
+            case didChange(UIPasteboard)
+            ///
+            ///  Contents of the pasteboard object have been removed.
+            ///
+            case didRemove(UIPasteboard)
+            
+        }
+        ///
+        /// Initializes a new `PasteboardMonitor`.
+        ///
+        /// - Parameters:
+        ///   - pasteboard:   The pasteboard to monitor.
+        ///   - queue:      The operation queue on which the handler executes. By
+        ///                 default, the main operation queue is used.
+        ///   - handler:    The handler to call when the state of the document
+        ///                 changes.
+        ///
+        
+        public init(pasteboard: UIPasteboard,
```

Comment:
Since this monitor wraps more than one notification, there needs to be an `options` parameter to allow the caller to select which events to monitor. See `KeyboardMonitor` for an example.